libsplinter/src/circuit/template/yaml_parser/mod.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod v0_1;
+
+use std::fs::File;
+use std::io::Read;
+
+use super::Error;
+
+#[derive(Deserialize, Debug)]
+struct TemplateVersionGuard {
+    version: String,
+}
+
+pub fn load_template(file_path: &str) -> Result<Template, Error> {
+    let mut file = File::open(file_path)
+        .map_err(|err| Error::new_with_source("Error opening template file", err.into()))?;
+    let mut data = Vec::new();
+    file.read_to_end(&mut data).map_err(|err| {
+        Error::new_with_source("Error reading data from template file", err.into())
+    })?;
+
+    let version_guard: TemplateVersionGuard = serde_yaml::from_slice(&data)?;
+    let template = Template::deserialize(&data, &version_guard.version)?;
+    Ok(template)
+}
+
+#[derive(Deserialize, Debug)]
+pub enum Template {
```

Comment:
Not sure about this yet, but we may explore using these templates with circuit update as well (trying to figure out how to apply the template to an existing circuit in order to add an additional node for example). Which makes these more "circuit templates" than "circuit create templates".

Reply:
I left as template for now. I understand  the argument for naming it `CircuitCreateTemplate` but also I think it might be confusing. The ` yaml_parser::v0_1::CircuitCreateTemplate` maps to the `rules::CircuitCreateTemplate`, so it makes sense to have the same name. But this `Template` struct is a wrapper over the versioned `yaml_parse::v*::CircuitCreateTemplate`, so they are not the same and naming it the same might be misleading.

Reply:
Sorry, just saw Shawn's comment. I'll rename it to CircuitTemplate. It's clearer than just `Template` and avoids any confusion `CircuitCreateTemplate` might cause